### PR TITLE
feat: various feats inc. label copy, fine-grained reactivity

### PR DESCRIPTION
**Pull request by cabreraalex**
Fix EOT-31

### CODE REVIEW

File: src/lib/components/Instances/InstanceTableRow.svelte [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
+		removeInstance: (id: number) => void;
+	}>();
 
 	let localInstanceInput = $state(instance.input);
 	let localInstanceLabel = $state(instance.label);
 	let rowHovered = $state(false);
 
 	let prediction = $derived(getPrediction(prompt, instance.id, instance.input, browser));
-	let metric = $derived(getMetric(prompt, instance, prediction));
+	let metric = $derived(getMetric(prompt, instance.label, prediction));
```
Comment (cabreraalex) [inline src/lib/components/Instances/InstanceTableRow.svelte:21]: This was running unnecessarily before

File: src/lib/components/Instances/InstanceTableRow.svelte [hunk 2]
```diff
@@ -51,33 +45,53 @@
 		contenteditable="plaintext-only"
 		class="box-border p-3"
 		bind:innerText={localInstanceInput}
-		onblur={() => changeInstance({ ...instance, input: localInstanceInput })}
+		onblur={() => {
```
Comment (cabreraalex) [inline src/lib/components/Instances/InstanceTableRow.svelte:48]: updated these so fine-grained reactivity works. Reassigning in parent caused whole thing to re-render and update.

File: src/lib/components/Instances/InstanceTableRow.svelte [hunk 3]
```diff
@@ -51,33 +45,53 @@
...
 	<td class="p-3">
 		{#await prediction}
 			Loading...
 		{:then pred}
 			{pred?.prediction}
 		{/await}
 	</td>
+	{#if project.show_labels}
+		<td class="relative h-full">
+			<button
```
Comment (cabreraalex) [inline src/lib/components/Instances/InstanceTableRow.svelte:68]: I'm mid on this. I think it's useful but a bit ugly/in the way.

File: src/lib/components/prompt/PromptBar.svelte [hunk 4]
```diff
@@ -27,7 +27,7 @@
 	}
 </script>
 
-<div class="flex h-full w-[450px] shrink-0 flex-col border-r px-6 py-4">
+<div class="flex h-full w-1/3 min-w-[300px] flex-col border-r px-6 py-4">
```
Comment (cabreraalex) [inline src/lib/components/prompt/PromptBar.svelte:30]: I think it's better a bit bigger.

**Comment by cabreraalex** [comment]
The overflows are still super broken sadly. Same with metric showing Naan all the time.

**Comment by Sparkier** [comment]
> The overflows are still super broken sadly.

where?
2024-02-14 at 09 21 52

File: src/lib/components/prompt/PromptBar.svelte (hunk 4 above)
Comment (Sparkier) [reply src/lib/components/prompt/PromptBar.svelte:30]: I think it looks pretty weird tbh. Reverting for now. I'd rather have the option to maximize it. Now that it's wider than tall on most screens looks weird to me.

File: src/lib/components/Instances/InstanceTableRow.svelte (hunk 3 above)
Comment (Sparkier) [reply src/lib/components/Instances/InstanceTableRow.svelte]: I changed the style. Merging for now to avoid conflicts but happy to reverse.

**Review by Sparkier — APPROVED** [review]

**Pull request merged**